sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffleCoalesceExec.scala

```diff
@@ -194,13 +205,80 @@ class JCudfTableOperator extends SerializedTableOperator[SerializedTableColumn]
   }
 }
 
+case class KudoHostMergeResultWrapper(inner: KudoHostMergeResult,
+    dataSize: Long) extends CoalescedHostResult {
+
+  /** Convert itself to a GPU batch */
+  override def toGpuBatch(dataTypes: Array[DataType]): ColumnarBatch = {
+    RmmRapidsRetryIterator.withRetryNoSplit {
+      closeOnExcept(inner.toTable) { cudfTable =>
+        GpuColumnVector.from(cudfTable, dataTypes)
+      }
+    }
+  }
+
+  /** Get the data size */
+  override def getDataSize: Long = dataSize
+
+  override def close(): Unit = inner.close()
+}
+
+case class RowCountOnlyMergeResult(rowCount: Int) extends CoalescedHostResult {
```

Comment:
NIT:
I would prefer moving the rows only result logic into the `KudoHostMergeResultWrapper`, instead of creating a new case class.
e.g. `case class KudoHostMergeResultWrapper(inner: Option[KudoHostMergeResult], rowsNum: Long)` or 
`case class KudoHostMergeResultWrapper(inner: Either[KudoHostMergeResult, Long])`